upup/pkg/fi/cloudup/gce/gce_cloud.go

```diff
@@ -29,24 +30,41 @@ import (
 	"k8s.io/kubernetes/federation/pkg/dnsprovider/providers/google/clouddns"
 )
 
-type GCECloud struct {
-	Compute *compute.Service
-	Storage *storage.Service
+type GCECloud interface {
```

Comment:
@andrewsykim I am thinking that we only use the cloud interface where it makes sense.  I am on the fence about this a bit the more I mull it over.  I do not want to make a method in an interface that gcloud does not use, for instance.

Let me push some changes, and have you take a look.